runtime/sys.c

```diff
@@ -640,6 +640,14 @@ CAMLprim value caml_sys_const_backend_type(value unit)
 {
   return Val_int(1); /* Bytecode backed */
 }
+CAMLprim value caml_sys_frame_pointers(value unit)
+{
+#if defined(NATIVE_CODE) && defined(WITH_FRAME_POINTERS)
+  return Val_true;
+#else
+  return Val_false;
+#endif
+}
```

Comment:
There's no need for this primitive. Edit `stdlib/Makefile` instead:
```diff
--- a/stdlib/Makefile
+++ b/stdlib/Makefile
@@ -180,8 +180,11 @@ stdlib.cma: $(OBJS)
 stdlib.cmxa: $(OBJS:.cmo=.cmx)
        $(CAMLOPT) -a -o $@ $^

-sys.ml: $(ROOTDIR)/VERSION sys.mlp
-       sed -e "s|%%VERSION%%|`sed -e 1q $< | tr -d '\r'`|" sys.mlp > $@
+VERSION = $(shell sed -e 1q $(ROOTDIR)/VERSION | tr -d '\r')
+sys.ml: sys.mlp $(ROOTDIR)/VERSION
+       sed $(call SUBST_STRING,VERSION) \
+           $(call SUBST,WITH_FRAME_POINTERS) \
+           $< > $@

 .PHONY: clean
 clean::
```

Reply:
Hmm.  Is adding complexity to the `Makefile` better than adding code in C and OCaml?  I'm unsure that it is.  What about the dune build?

Reply:
This is how `utils/config.ml` is generated - it adds no complexity at all... the functions are already there. Dune is irrelevant to the consideration (FWIW, it would require a switch for stdlib/sys.ml to be generated in the same way config.ml, which happens to be by calling make).

Reply:
>Is adding complexity to the `Makefile` better than adding code in C and OCaml?

We are big fans of doing things statically rather than dynamically so yes, definitely. Especially when all the machinery is already available.